src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -6848,7 +6848,7 @@ VOID MethodTableBuilder::ValidateInterfaceMethodConstraints()
 
             // Grab the method token
             MethodTable * pMTItf = pItf->GetMethodTable();
-            CONSISTENCY_CHECK(CheckPointer(pMTItf->GetMethodDescForSlot(it.GetSlotNumber())));
+            CONSISTENCY_CHECK(CheckPointer(pMTItf->GetMethodDescForSlot_NoThrow(it.GetSlotNumber())));
```

Comment:
The `STANDARD_VM_CONTRACT` permits throwing. This seems odd.

Reply:
This is an assert. The `GetMethodDescForSlot` and `GetMethodDescForSlot_NoThrow` are designed to produce exactly the same results, given the same inputs. The only difference is in performance of repeated calls, and the fact that GetMethodDescForSlot has some side effects on the slot data structures. There is no reason to avoid the _NoThrow variant in asserts, and since the _NoThrow variant doesn't have various side effects, it is preferable to use the `GetMethodDescForSlot_NoThrow` function when in code which is only used in DEBUG code paths.